packages/transaction-controller/src/TransactionController.ts

```diff
@@ -2365,20 +2698,20 @@ export class TransactionController extends BaseControllerV1<
   ): TransactionMeta[] {
     const nonceNetworkSet = new Set();
 
-    const txsToKeep = transactions
+    const txsToKeep = [...transactions]
       .sort((a, b) => (a.time > b.time ? -1 : 1)) // Descending time order
       .filter((tx) => {
         const { chainId, status, txParams, time } = tx;
 
         if (txParams) {
-          const key = `${txParams.nonce}-${convertHexToDecimal(
-            chainId,
+          const key = `${String(txParams.nonce)}-${String(
+            convertHexToDecimal(chainId),
```

Comment:
is there a reason why we are so explicit here?

Reply:
Good point. We need it for `txParams.nonce` because it could be `undefined` — we get a lint warning if we try to take that away — but we don't need it for the `convertHexToDecimal` call. Removed here: [06fae6d](https://github.com/MetaMask/core/pull/3827/commits/06fae6d1165fd34dd60ad25ca86484bf900ba796)